eclipse-collections-code-generator/src/main/resources/impl/list/immutable/immutablePrimitiveSingletonList.stg

```diff
@@ -397,6 +402,32 @@ final class Immutable<name>SingletonList implements Immutable<name>List, Seriali
     {
         throw new UnsupportedOperationException("subList not yet implemented!");
     }
+
+    /**
+     * @since 9.1.
+     */
+    @Override
+    public ImmutableList\<<name><name>Pair> zip<name>(<name>List list)
+    {
+        if (list.isEmpty())
+        {
+            return Lists.immutable.empty();
+        }
+        return Lists.immutable.with(PrimitiveTuples.pair(this.get(0), list.get(0)));
```

Comment:
this.get(0) should be replaced by this.element1

Reply:
Makes sense.  Fixed.